src/ObjectSchema.js

```diff
@@ -323,7 +324,9 @@ const ObjectSchema = ({ schema = initialState, ...options } = {}) => {
           ...schema,
           [target]: [
             ...schema[target],
-            $ref ? { name, $ref } : Object.assign({}, { name }, attributes)
+            $ref
+              ? (Object.keys(restAttributes).length ? { name, $ref, ...restAttributes } : { name, $ref })
```

Comment:
Why not

```suggestion
              ? { name, $ref, ...restAttributes }
```

Reply:
The distinction makes no difference imho. If the object has no keys then spreading them will not add any key.

Reply:
Good catch updated to suggested change